org.alloytools.alloy.application/src/main/java/edu/mit/csail/sdg/alloy4viz/VizGUI.java

```diff
@@ -1305,15 +1457,121 @@ private Runner doExportXml() {
         return null;
     }
 
+    // [HASLab]
+    // ad hoc implementation since alloy lacks a proper pretty printer
+    // also, prints conjunctions as lists, which can't be parsed
+    private Runner doExportLTL() {
```

Comment:
this is an improvised method to export an instance as a formula after several user requests; it's rather ad hoc the default printing of expressions is not parsable